sdk/src/main/java/velocitas/sdk/VehicleApp.kt

```diff
@@ -0,0 +1,47 @@
+package velocitas.sdk
+
+import velocitas.sdk.middleware.Middleware
+
+/**
+ * Base class for all vehicle apps which manages an app's lifecycle.
+ */
+abstract class VehicleApp
+{
+
+    /**
+     * Runs the Vehicle App.
+     */
+    fun run() {
+        Logger.info("Running App...")
+        Middleware.getInstance().start()
```

Comment:
> Probably from the c++ variant taken ...

Yes: https://github.com/eclipse-velocitas/vehicle-app-cpp-sdk/blob/main/sdk/src/sdk/VehicleApp.cpp#L41

> ... but can we do it better?

Since this would be a breaking change, this is to be discussed with the team

> ... If yes I am wondering why we are not using the SingletonHolder pattern that was introduced here. :)

That's a valid point :D

Reply:
Implemented a MiddlewareFactory which will instantiate the Middleware